### DISCUSSION TITLE
RTU over a TCP connexion with a gateway

### DISCUSSION THREAD

**Discussion by waedg** [rossmann-engineering/EasyModbusTCP.NET#64] [Ideas]
Hi floks, 
Communication with an RTU device over a gateway is not supported. Any hints to add that feature ?

**Comment by Padanian** [comment]
What gateway? A tcp to rtu gateway?

**Comment by lantosno** [comment]
Hello
The author probably means the MODBUS TCP gateway (For example, MOXA MB3180). Where the client connects to the MB3180 via MODBUS TCP, and several slaves are connected via MODBUS RTU.
For example, I want to access the address 192.168.1.5 and the client on it with the number 247. The request should be something like this: Client.ReadInputRegisters(Slave_Number=247,Start_Adress,Quantity). 
The result of this function is 00 CD 00 00 00 06 01 04 00 F7 00 0C. BUT the library only outputs 00 01 00 00 00 06 01 04 00 F7 00 0C.

**Comment by Padanian** [comment]
Wow, necroposting at its best. The question was 4 years old.
